### ISSUE TITLE
Fleet cannot update repo:  ssh: Could not resolve hostname bitbucket.org

### ISSUE THREAD

**Issue by Shaked** [rancher/fleet#915]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Once every few days we encounter all git repositories showing this error:

> time="2022-08-16T07:35:22Z" level=error msg="bundle gitrepos-devices-vcluster-main-gitrepos-devices: gitrepo.fleet.cattle.io fleet-default/docker-registry error] git ls-remote git@bitbucket.org:......git refs/heads/master error: exit status 128, detail: ssh: Could not resolve hostname bitbucket.org: Try again\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists."

This happens out of nowhere, while no permissions were changed and bitbucket.org is accessible from the cluster, from local machine and doesn't report any issues in https://bitbucket.status.atlassian.com/.

Only after I restart the `gijob` pod i.e `kubectl delete pod -n fleet-system gitjob....` everything starts working again. 

### Expected Behavior

As long as bitbucket.org is available and permissions are set, this error should not happen. 

### Steps To Reproduce

1. Install rancher 2.6.4 or 2.6.6
2. Create a `GitRepo` pointing to a bitbucket cloud repository 
3. Set an access key to that repository
4. Wait until this happens (no idea when/why)
5. Restart the `fleet-system/gitjob` pod to recover from this error 

### Environment

```markdown
- Architecture: x86
- Fleet Version: 0.3.9
- Cluster: 
  - Provider: k3s
  - Options:
  - Kubernetes Version: `v1.21.12+k3s1` and `v1.20.15+k3s1` (it happened on two different clusters)
```

### Anything else?

![gitrepos-blur](https://user-images.githubusercontent.com/960015/184849121-7e564bb0-85fd-49ae-bbce-dcb3869feb62.jpeg)
[test.log](https://github.com/rancher/fleet/files/9349579/test.log)

**Comment by thardeck** [comment]
Related to #1118

**Comment by thardeck** [comment]
@Shaked we have added a fix for the issue and released https://github.com/rancher/fleet/releases/tag/v0.6.0-alpha3 . Could you verify that it fixes your issue?
We will also most likely backport the fix to older Fleet versions which will be part of future rancher releases.

**Comment by Shaked** [comment]
@thardeck honestly I am not sure why it happened up until now so it would be hard to verify the fix. Having said that I will keep an eye on it once we upgrade to the latest rancher & fleet version.

**Comment by kkaempf** [comment]
fixed in 0.6.0
